### ISSUE TITLE
stream.respond() not ordering :status header correctly

### ISSUE THREAD

**Issue by mcollina** [nodejs/http2#40]
This works:

```js
 function onStream (stream) {
  stream.respond({
    ':status': 200,
    'content-type': 'text/html'
  })
  stream.end('<html><head></head><body><h1>this is some data</h2></body></html>');
}
```

This doesn't:

```js
function onStream (stream) {
  stream.respond({
    'content-type': 'text/html',
    ':status': 200
  })
  stream.end('<html><head></head><body><h1>this is some data</h2></body></html>');
}
```

This is the output

```
$ nghttp -v http://localhost:8000/hello
[  0.003] Connected
[  0.003] send SETTINGS frame <length=12, flags=0x00, stream_id=0>
          (niv=2)
          [SETTINGS_MAX_CONCURRENT_STREAMS(0x03):100]
          [SETTINGS_INITIAL_WINDOW_SIZE(0x04):65535]
[  0.003] send PRIORITY frame <length=5, flags=0x00, stream_id=3>
          (dep_stream_id=0, weight=201, exclusive=0)
[  0.003] send PRIORITY frame <length=5, flags=0x00, stream_id=5>
          (dep_stream_id=0, weight=101, exclusive=0)
[  0.003] send PRIORITY frame <length=5, flags=0x00, stream_id=7>
          (dep_stream_id=0, weight=1, exclusive=0)
[  0.003] send PRIORITY frame <length=5, flags=0x00, stream_id=9>
          (dep_stream_id=7, weight=1, exclusive=0)
[  0.003] send PRIORITY frame <length=5, flags=0x00, stream_id=11>
          (dep_stream_id=3, weight=1, exclusive=0)
[  0.003] send HEADERS frame <length=44, flags=0x25, stream_id=13>
          ; END_STREAM | END_HEADERS | PRIORITY
          (padlen=0, dep_stream_id=11, weight=16, exclusive=0)
          ; Open new stream
          :method: GET
          :path: /hello
          :scheme: http
          :authority: localhost:8000
          accept: */*
          accept-encoding: gzip, deflate
          user-agent: nghttp2/1.17.0
[  0.007] recv SETTINGS frame <length=0, flags=0x00, stream_id=0>
          (niv=0)
[  0.007] recv SETTINGS frame <length=0, flags=0x01, stream_id=0>
          ; ACK
          (niv=0)
[  0.007] send SETTINGS frame <length=0, flags=0x01, stream_id=0>
          ; ACK
          (niv=0)
[  0.016] recv (stream_id=13) content-type: text/html
[  0.016] [ERROR] Invalid HTTP header field was received: frame type: 1, stream: 13, name: [:status], value: [200]
[  0.016] [INVALID; error=Invalid HTTP header field was received] recv HEADERS frame <length=34, flags=0x04, stream_id=13>
          ; END_HEADERS
          (padlen=0)
          ; First response header
[  0.016] send RST_STREAM frame <length=4, flags=0x00, stream_id=13>
          (error_code=PROTOCOL_ERROR(0x01))
[  0.016] send GOAWAY frame <length=8, flags=0x00, stream_id=0>
          (last_stream_id=0, error_code=NO_ERROR(0x00), opaque_data(0)=[])
Some requests were not processed. total=1, processed=0
```

cc @jasnell I think we should send `:status` before everything

**Comment by jasnell** [comment]
There is a fix for this in my round 10 pr

**Comment by mcollina** [comment]
awesome. I'll need to get #37 rebased when that lands.

**Comment by mcollina** [comment]
Fixed by #36
